### PR TITLE
Remove dependency on Jetpack from Shipping and Tax task list items

### PR DESCRIPTION
**Pull request by waclawjacek**
### Changes proposed in this Pull Request:

WooCommerce Shipping & Tax will soon no longer require Jetpack (paJDVv-2t2-p2).

As part of this change, this PR updates the WC Home tasks to not install (or mention) Jetpack alongside WooCommerce Shipping & Tax.

Part of https://github.com/woocommerce/shipping/issues/15

#### What's changed by this PR

Jetpack-the-plugin is no longer installed by either:

1) the shipping and tax tasks on WC Home:

image

2) the `ExperimentalShippingRecommendation` feature which is the banner displayed on e.g. the WC Settings > Shipping page:

image

Connection to WPCOM infrastructure is now managed by building an auth URL using the onboarding data store and checked by interfacing with the Jetpack Connection PHP package which is bundled in WC Core rather than through Jetpack-the-plugin. Since the WCS&T extension itself will also use Jetpack Connection and not require Jetpack-the-plugin, mentions of Jetpack are removed or replaced with mentions of WPCOM to reflect this.

The WPCOM connection flow used is now Woo-branded (previously: Jetpack-branded) but only when connecting from the task list.

#### How does this PR achieve the change

- Removes the requirement to install and mentions of Jetpack-the-plugin from WCS&T-related onboarding.
- Uses the Onboarding data store's `getJetpackAuthUrl()` function to generate the Jetpack auth link.
- Interfaces with the Jetpack Connection package (rather than Jetpack-the-plugin) to check if Jetpack is connected (site + user token are both required).

### How to test the changes in this Pull Request:

Things to test (each described in more detail in its own section):

<details>
<summary>0) Test site setup (prerequisite; repeat this for every test below)</summary>

1. Ensure you have your WPCOM sandbox running, `trunk` checked out and the D63626-code diff applied (`arc patch D63626`) so you can test the Woo-branded onboarding design.
1. Set up a new clean test site (no Woo, no Jetpack).
1. Go to WP admin > Plugins > Add New > Upload Plugin.
   1. First, let's build and install **but not activate** a version of WCS&T that is using Jetpack Connection instead of Jetpack-the-plugin. As that version is not yet merged as of writing this, let's compile this WCS&T PR https://github.com/Automattic/woocommerce-services/pull/2439 by running `nvm use && npm i && npm run build` and upload the resulting `woocommerce-services.zip` onto your test site. **Do not activate it!** We only upload and install it so that when the WC Core tasks try to get a copy of the WCS&T plugin, they will instead use the one we've just installed. This will allow us to test this WC Core PR without any quirks related to the dependency on Jetpack-the-plugin coming up.
   1. Then, let's build this PR. After checking out this repository and branch, generate a ZIP by running this in the repo root directory: `nvm use && pnpm i && pnpm run build && cd plugins/woocommerce && pnpm run build:zip`. Upload & activate the built plugin ZIP.
1. This will launch the Woo onboarding wizard. The changes in this PR affect the WC Home task lists and not this wizard so let's click "Skip guided setup" in the top right:
   image
1. This will ask you "Where is your business located?" Pick US California.
1. You will land on the WC Home page where the task lists dwell, ready to go through the test cases described below:
   image
1. Before we test though, let's create a sample physical product so we can check if WCS&T really is connected to WPCOM after we go through the onboarding tasks. A minimal example product can be:
   - Product name: Hoodie
   - Price: $45
   - Dimensions: 1x2x3
   - Weight: 1

</details>

<details>
<summary>1) WCS&T installation through the shipping task</summary>

0. Follow the instructions in the "0) Test site setup (repeat for every test below)" section above.
1. image
1. image
1. image
7. image
8. image
9. Return here with WCS&T installed, WPCOM connected (but no Jetpack installed), and task marked as completed.
   image
1. Create a new order with your sample physical product and verify WCS&T's "Create shipping label" section is presented when you visit that order's page in WP Admin.

</details>

<details>
<summary>2) WCS&T installation through the tax task</summary>

0. Follow the instructions in the "0) Test site setup (repeat for every test below)" section above.
1. image
3. image
4. image
5. Screenshot 2023-09-29 at 13-22-46 Home ‹ WooCommerce ‹ Fulfill — WooCommerce
6. image
7. image
8. image
9. At this stage, WooCommerce Shipping & Tax is installed and Jetpack-the-plugin isn't. Tax calculations in WC Settings are still disabled:
   image
11. image
12. Return to the WC Home page with WCS&T installed, WPCOM connected but Jetpack not installed, and the tax task marked as completed.
13. Tax calculations are now enabled. You can test this by adding a product to the cart and inputting an actual address somewhere within the US on the shopper-facing cart/checkout page.

</details>

<details>
<summary>3) WCS&T installation through the "experimental shipping task" (banner on the WC Settings > Shipping page)</summary>

0. Follow the instructions in the "0) Test site setup (repeat for every test below)" section above.
1. This time we don't do anything on the WC Home page. Instead, go to WC Settings > Shipping and at the bottom see this banner. Click "Get started":
   image
3. image
4. image
5. image
6. image
7. image
8. Screen Shot 2023-09-29 at 13 58 42-fullpage
9. Return here with WCS&T installed, WPCOM connected but Jetpack not installed, and the task no longer showing on the WC Settings > Shipping page.
   image

1. Create a new order with your sample physical product and verify WCS&T's "Create shipping label" section is presented when you visit that order's page in WP Admin.

</details>

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Remove dependency on Jetpack from WooCommerce Shipping & Tax onboarding tasks

#### Comment 

</details>

### CODE REVIEW

File: packages/js/components/src/plugins/index.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ export const Plugins = ( {
 	onAbort,
 	onComplete,
 	onError = () => null,
-	pluginSlugs = [ 'jetpack', 'woocommerce-services' ],
+	pluginSlugs = [ 'woocommerce-services' ],
```
Comment (waclawjacek) [inline packages/js/components/src/plugins/index.tsx:40]: This is a default for all instances of `<Plugins>`. I've updated the code in https://github.com/woocommerce/woocommerce/pull/39992/commits/602c5e2e903942bb29cdaee23eec569d33e0afad so each task using `<Plugins>` (only shipping & tax anyway) explicitly specifies the plugin slugs.

I've kept this optional not to disrupt too much but removed `'jetpack'` as WCS&T won't require Jetpack anymore.

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import { Component, Fragment } from '@wordpress/element';
 import { compose } from '@wordpress/compose';
 import PropTypes from 'prop-types';
 import { withDispatch, withSelect } from '@wordpress/data';
-import { PLUGINS_STORE_NAME } from '@woocommerce/data';
+import { ONBOARDING_STORE_NAME } from '@woocommerce/data';
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js:10]: I am not sure if it is a good idea to use the "onboarding" data store. It works great this way but I am not sure if by using something that is meant for "onboarding" (so in my understanding: the fresh install setup wizard) there	won't be any side effects/confusion?

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ export class Connect extends Component {
 	}
 
 	async connectJetpack() {
-		const { jetpackConnectUrl, onConnect } = this.props;
+		const { jetpackAuthUrl, onConnect } = this.props;
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js]: The onboarding data store calls this `jetpackAuthUrl` so I stuck with that naming here.

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 4]
```diff
@@ -114,15 +114,15 @@ Connect.propTypes = {
 	 */
 	isRequesting: PropTypes.bool,
 	/**
-	 * Generated Jetpack connection URL.
+	 * Generated Jetpack authentication URL.
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js:115]: Again, only because this is what the onboarding data store calls it.

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 5]
```diff
@@ -157,20 +157,24 @@ Connect.defaultProps = {
...
-			redirect_url: props.redirectUrl || window.location.href,
+			redirectUrl: props.redirectUrl || window.location.href,
+			from: 'woocommerce-services',
 		};
-		const isRequesting = isPluginsRequesting( 'getJetpackConnectUrl' );
-		const error = getPluginsError( 'getJetpackConnectUrl' ) || '';
-		const jetpackConnectUrl = getJetpackConnectUrl( queryArgs );
+		const isRequesting = isOnboardingRequesting( 'getJetpackAuthUrl' );
+		const error = getOnboardingError( 'getJetpackAuthUrl' ) || '';
+		const jetpackAuthUrl = getJetpackAuthUrl( queryArgs ).url;
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js]: This was a really easy transition to do between the Plugins and Onboarding stores but I noticed that the plugins data store returned a string for `jetpackConnectUrl`, whereas the onboarding data store returns an object informing of the error, etc., so we have to extract the `.url` prop here.

I am wondering if it is OK to ignore the other fields in the response object indicating request success etc. and just extract the `.url` (which seems to always be there), provided we check the values of `isRequesting` and `error` as defined here?

File: plugins/woocommerce-admin/client/shipping/woocommerce-services-item.tsx [hunk 6]
```diff
@@ -31,7 +31,7 @@ const WooCommerceServicesItem: React.FC< {
 				actions.push( {
 					url: getAdminLink( 'plugins.php' ),
 					label: __(
-						'Finish the setup by connecting your store to Jetpack.',
+						'Finish the setup by connecting your store to WordPress.com.',
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/shipping/woocommerce-services-item.tsx:34]: Consistent with https://github.com/Automattic/woocommerce-services/pull/2677#issue-1871547685

File: plugins/woocommerce-admin/client/task-lists/fills/tax/woocommerce-tax/card.tsx [hunk 7]
```diff
@@ -31,29 +31,12 @@ export const Card: React.FC< TaxChildProps > = () => {
...
-							<Link
-								type="external"
-								href="https://woocommerce.com/products/jetpack/?utm_medium=product"
-								target="_blank"
-							>
-								<></>
-							</Link>
-						),
-					},
-				} ),
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/task-lists/fills/tax/woocommerce-tax/card.tsx:50]: I've removed the bullet item about Jetpack because we're not installing it when installing WCS&T anymore.

Screenshot 2023-09-04 at 13 04 03

The ToS item is updated several lines later.

File: plugins/woocommerce-admin/docs/features/onboarding.md [hunk 8]
```diff
@@ -72,20 +72,20 @@ Both of these endpoints use WooCommerce Core's `WC_Helper_API` directly. The mai
 
 To disconnect from WooCommerce.com, go to `WooCommerce > Extensions > WooCommerce.com Subscriptions > Connected to WooCommerce.com > Disconnect`.
 
-## Jetpack Connection
+## WordPress.com Connection
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/docs/features/onboarding.md:75]: Naming in line with https://github.com/Automattic/woocommerce-services/pull/2677#issue-1871547685

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/ExperimentalShippingRecommendation.php [hunk 9]
```diff
@@ -89,9 +89,8 @@ public static function has_plugins_active() {
 	 * @return bool
 	 */
 	public static function has_jetpack_connected() {
-		if ( class_exists( '\Jetpack' ) && is_callable( '\Jetpack::is_connection_ready' ) ) {
-			return \Jetpack::is_connection_ready();
-		}
-		return false;
+		$jetpack_connection_manager = new Manager( 'woocommerce' );
+
+		return $jetpack_connection_manager->is_connected() && $jetpack_connection_manager->has_connected_owner();
```
Comment (waclawjacek) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/ExperimentalShippingRecommendation.php:94]: Use Jetpack Connection's `Manager` class for managing the connection.

I used the `woocommerce` plugin slug because that's what the WC onboarding uses. This can be anything AFAIK; preferably the plugin being installed but since it isn't installed yet, I thought I should use whatever WC Core uses, which either `woocommerce` or `null`.

In this use case, it might not affect anything.

File: plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php [hunk 10]
```diff
@@ -79,7 +79,7 @@ public function get_title() {
 	public function get_content() {
 		return self::can_use_automated_taxes()
 			? __(
-				'Good news! WooCommerce Services and Jetpack can automate your sales tax calculations for you.',
+				'Good news! WooCommerce Tax can automate your sales tax calculations for you.',
```
Comment (waclawjacek) [inline plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Tax.php:82]: This isn't really visible anywhere aside from the API response. I figured let's change it just in case these become visible in the future.

Markup on 2023-09-06 at 00:16:49

File: plugins/woocommerce-admin/client/task-lists/setup-task-list/components/task-headers/tax.js [hunk 11]
```diff
@@ -22,7 +22,7 @@ const TaxHeader = ( { task, goToTask } ) => {
 				<h1>{ __( 'Add your tax rates', 'woocommerce' ) }</h1>
 				<p>
 					{ __(
-						'Set up tax rates manually or use WooCommerce and Jetpack to automate your sales tax calculations for you.',
+						'Set up tax rates manually or use WooCommerce Tax to automate your sales tax calculations for you.',
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/task-lists/setup-task-list/components/task-headers/tax.js:25]: image

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 12]
```diff
@@ -157,20 +157,24 @@ Connect.defaultProps = {
...
+		const {
+			getJetpackAuthUrl,
+			isOnboardingRequesting,
+			getOnboardingError,
+		} = select( ONBOARDING_STORE_NAME );
 
 		const queryArgs = {
-			redirect_url: props.redirectUrl || window.location.href,
+			redirectUrl: props.redirectUrl || window.location.href,
+			from: 'woocommerce-services',
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js:164]: `woocommerce-services` means "go to a Woo-branded flow" as demonstrated here: https://github.com/Automattic/wp-calypso/pull/35193

One alternative is Jetpack + Woo co-branding (value: `woocommerce-onboarding`) but we no longer require Jetpack: https://github.com/Automattic/wp-calypso/pull/34380

File: plugins/woocommerce-admin/client/task-lists/fills/experimental-shipping-recommendation/test/shipping-recommendation.tsx [hunk 13]
```diff
@@ -60,51 +60,38 @@ const ShippingRecommendation = ( props: ShippingRecommendationProps ) => {
 };
 
 describe( 'ShippingRecommendation', () => {
-	test( 'should show plugins step when jetpack is not installed and activated', () => {
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/task-lists/fills/experimental-shipping-recommendation/test/shipping-recommendation.tsx:63]: We don't install Jetpack now so I am removing this test case.

File: plugins/woocommerce-admin/client/task-lists/fills/tax/woocommerce-tax/plugins.tsx [hunk 14]
```diff
@@ -99,16 +89,21 @@ export const Plugins: React.FC< SetupStepProps > = ( {
...
 				abortText={ __( "I don't charge sales tax", 'woocommerce' ) }
+				pluginSlugs={ pluginsToActivate }
 			/>
 			{ ! tosAccepted && (
 				<Text
 					variant="caption"
 					className="woocommerce-task__caption"
 					size="12"
 					lineHeight="16px"
+					style={ { display: 'block' } }
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/task-lists/fills/tax/woocommerce-tax/plugins.tsx]: This is a bugfix that I'm sneaking in here.

Previously:

Screenshot 2023-09-29 at 13-25-36 Home ‹ WooCommerce ‹ Fulfill — WooCommerce

With this fix:

Screenshot 2023-09-29 at 13-22-46 Home ‹ WooCommerce ‹ Fulfill — WooCommerce

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 15]
```diff
@@ -47,14 +47,14 @@ export class Connect extends Component {
...
-				window.location = jetpackConnectUrl;
+				window.location = jetpackAuthUrl;
 			}
 		);
 	}
 
 	render() {
 		const {
-			hasErrors,
+			error,
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js]: This is a drive-by fix. I noticed there are no references to a prop called `hasErrors` here or where this `<Connect>` component is used. I removed this prop and replaced it with a check on the `error` prop. Tested and it works.

Previously, when `error` would be set, nothing would happen.

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 16]
```diff
@@ -181,3 +164,20 @@ export default compose(
...
+	);
+
+	const queryArgs = {
+		redirectUrl: redirectUrl || window.location.href,
+		from: 'woocommerce-services',
+	};
+
+	return {
+		error: getOnboardingError( 'getJetpackAuthUrl' ) || '',
+		isRequesting: isResolving( 'getJetpackAuthUrl', [ queryArgs ] ),
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js]: The `getJetpackAuthUrl` function from the Onboarding data store does not use a `setIsRequesting` call like the function from the Plugins data store used to so we use the `@wordpress/data`-generic `isResolving` instead.

File: plugins/woocommerce-admin/client/task-lists/fills/tax/woocommerce-tax/setup.tsx [hunk 17]
```diff
@@ -91,6 +96,10 @@ export const Setup: React.FC< SetupProps > = ( {
 		pluginsToActivate,
 	};
 
+	const prefetchedJetpackAuthData = useSelect( ( select ) => {
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/task-lists/fills/tax/woocommerce-tax/setup.tsx:99]: Prefetching the Jetpack auth URL here and then passing it to the `<Connect>` component so users don't have to wait for the "Connect" to finish retrieving the auth URL from the REST API.

File: plugins/woocommerce-admin/client/task-lists/fills/shipping/index.js [hunk 18]
```diff
@@ -687,6 +672,11 @@ const ShippingWrapper = compose(
 		const countryName = country ? country.name : null;
 		const activePlugins = getActivePlugins();
 
+		const prefetchedJetpackAuthData = prefetchJetpackAuthData(
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/task-lists/fills/shipping/index.js:675]: Prefetching the Jetpack auth URL here and then passing it to the `<Connect>` component so users don't have to wait for the "Connect" to finish retrieving the auth URL from the REST API.

File: plugins/woocommerce-admin/client/task-lists/fills/experimental-shipping-recommendation/shipping-recommendation.tsx [hunk 19]
```diff
@@ -76,6 +78,10 @@ export const ShippingRecommendation: React.FC<
 		setPluginsToActivate( remainingPlugins );
 	}, [ activePlugins, isJetpackConnected, isResolving, pluginsToActivate ] );
 
+	const prefetchedJetpackAuthData = useSelect( ( select ) => {
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/task-lists/fills/experimental-shipping-recommendation/shipping-recommendation.tsx:81]: Prefetching the Jetpack auth URL here and then passing it to the `<Connect>` component so users don't have to wait for the "Connect" to finish retrieving the auth URL from the REST API.

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 20]
```diff
@@ -21,8 +21,12 @@ export class Connect extends Component {
 	}
 
 	componentDidUpdate( prevProps ) {
-		const { createNotice, error, isRequesting, onError, setIsPending } =
-			this.props;
+		const {
+			createNotice,
+			jetpackAuth: { error, isRequesting },
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js:26]: Previously the component would figure out the URL (alongside the error and "is retrieving" status) internally. Now it receives it as a prop. This allows us to prefetch the URL which allows us to immediately present a clickable "Connect" button to users (vs. requiring them to wait until the REST API responds with a formatted Jetpack auth URL if we started retrieving that URL once the Connect button is rendered).

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 21]
```diff
@@ -155,22 +147,23 @@ Connect.defaultProps = {
 	setIsPending: () => {},
 };
 
+const formatJetpackAuthUrlQueryArgs = ( redirectUrl ) => ( {
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js]: Extracted this here so that both the Connect component and the PrefetchJetpackAuthUrl construct the query args the same way (required for prefetching to work correctly).

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 22]
```diff
@@ -181,3 +174,16 @@ export default compose(
 		};
 	} )
 )( Connect );
+
+export const PrefetchJetpackAuthUrl = ( { redirectUrl = undefined } ) => {
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js]: This component allows us to prefetch the Jetpack auth URL. Asking the data store for the URL is enough for it to cache it.

Without prefetching, upon arriving at the task step where the "Connect" button is displayed, a user going through the task would have to wait until the REST API responds with a Jetpack URL which could take some time.

With prefetching, there is no wait.

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js (hunk 2 above)
Comment (adrianduffell) [reply plugins/woocommerce-admin/client/dashboard/components/connect/index.js:10]: This is ok -- we consider the task list to be part of onboarding.

**Review by adrianduffell — APPROVED** [review]
Thanks @waclawjacek, this is looking really great!

I've left a suggestion for the prefetching, but otherwise I think this is almost ready.

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js (hunk 22 above)
Comment (adrianduffell) [reply plugins/woocommerce-admin/client/dashboard/components/connect/index.js]: Would it be possible to bring the prefetching to the `Setup` component as a `useEffect` hook? This would make it more consistent with the rest of WC Admin -- the `<PrefetchJetpackAuthUrl />` style component would be introducing a new pattern to the code base.

Comment (waclawjacek) [reply plugins/woocommerce-admin/client/dashboard/components/connect/index.js]: Done in https://github.com/woocommerce/woocommerce/compare/3801bff79f7f5bab43d40c301d543078dcfa529b...de347bcffbef34a0a931911d00e9f29fa77edc6f

**Comment by waclawjacek** [comment]
Thank you so much for the review, @adrianduffell! With the prefetching fix in place, would you like to give this PR one more look?

File: plugins/woocommerce-admin/client/dashboard/components/connect/index.js [hunk 23]
```diff
@@ -7,64 +7,65 @@ import { Component, Fragment } from '@wordpress/element';
...
+ * Only render this component when the user has accepted Jetpack's Terms of Service.
+ * The API endpoint used by this component sets "jetpack_tos_agreed" to true when
+ * returning the URL.
+ */
 export class Connect extends Component {
 	constructor( props ) {
 		super( props );
 		this.state = {
-			isConnecting: false,
+			isAwaitingRedirect: false,
```
Comment (waclawjacek) [inline plugins/woocommerce-admin/client/dashboard/components/connect/index.js:23]: Previously, this button would be disabled until it gets the URL. This used to be a quick operation but with the new endpoint, getting the URL from the API takes a while.

I tried resolving it by adding a prefetch operation that would retrieve the URL when the entire stepper is loaded. This worked but since accessing the `jetpack-authorization-url` endpoint also sets `jetpack_tos_agreed` to `true` in the database, I decided if we start prefetching immediately after the user accepts the TOS (= which is when we show the Connect button), we still speed things up a bit.

The button is no longer disabled when the URL is being loaded. Instead, it will register the fact that it was clicked, display a loading animation (`isBusy`) and only actually redirect once it gets the URL.

**Comment by waclawjacek** [comment]
@adrianduffell Regarding the prefetching issue we discussed in p1697115597594889-slack-C01SFMVEYAK, I have implemented a fix for it in 74039d1ec93c3738c1d2b194b23b4ec27180dee6.

I have described this new approach here: https://github.com/woocommerce/woocommerce/pull/39992#discussion_r1361834979

Could you please give this PR one more review and LMK if it's good to merge once that's done?

**Comment by adrianduffell** [comment]
Hi @waclawjacek -- the new code LGTM. I think it's ready to merge, but I noticed one thing that may need a quick first: I got the Jetpack branding when connecting from the Tax task 🤔

**Comment by waclawjacek** [comment]
@adrianduffell Thank you for reviewing! I've just tested and I got the Woo branding. Could it be that you didn't rewrite hosts on your local machine to send you to your WPCOM sandbox/didn't apply the WPCOM patch?

**Comment by adrianduffell** [comment]
> @adrianduffell Thank you for reviewing! I've just tested and I got the Woo branding. Could it be that you didn't rewrite hosts on your local machine to send you to your WPCOM sandbox/didn't apply the WPCOM patch?

@waclawjacek  Ah yes -- that's reason! Sorry, I missed that step in your instructions 😅 

I think this is all good to merge then!

**Comment by adrianduffell** [comment]
Merging this on behalf of @waclawjacek
